core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala

```diff
@@ -93,9 +93,10 @@ case class DeleteCommand(
       // Re-cache all cached plans(including this relation itself, if it's cached) that refer to
       // this data source relation.
       sparkSession.sharedState.cacheManager.recacheByPlan(sparkSession, target)
+      Seq(
+        Row(this.metrics.getOrElse("numDeletedRows", 0))
```

Comment:
nit: `Seq(Row(...))` (or better? `Row(...) :: Nil`

Reply:
@jaceklaskowski Can you clarify with me your view on how using the factory method defined on the Seq companion object compare with the concrete list constructor ? I would say that the factory method is preferrable from the point of view of general programming, as it can return a specific implementation of the trait depending on some properties of the input data.

I'll fix the formatting

Reply:
you can just use `metrics("numDeletedRows").value`. It is initialized with a default value of 0, anyways

Reply:
You should also cover the case of deletes at partition boundaries. Deletes at partition boundaries is a metadata operation, therefore we don't actually have any information around how many rows were deleted.

So if `metrics("numRemovedFiles")` is greater than 0, and `metrics("numDeletedRows")` is equal to 0, then we have reached this metadata-only delete case. in that case, let's just return -1 since we don't actually know how many rows were deleted.

Reply:
@scottsand-db 

Using the `get` method on the map, we get an option and we are kind of forced to deal with the potential absence of the value, which is more robust. Don't you think that accessing a map directly relying on the fact that same values are always present is relying on some implicit knowledge about the content of the map that is not encoded within the type system and which can also change leading to failures?

```scala
     Seq(Row(this.metrics.get("numDeletedRows").map(_.value).getOrElse(0L)))
```

Reply:
sure, that SGTM. make sure to cover the partition boundary case, too.